app/src/main/java/io/apicurio/registry/metrics/LivenessCheck.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2019 Red Hat
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.apicurio.registry.metrics;
+
+/**
+ * Common interface for a liveness check.
+ *
+ * @author Jakub Senko <jsenko@redhat.com>
+ */
+public interface LivenessCheck {
+
+    void suspect(String reason);
+
+    void suspectWithException(Throwable reason);
```

Comment:
Since the arguments are different, this could just be:

`void suspect(Throwable reason)`